modules/doobie/src/main/scala/DoobieMapping.scala

```diff
@@ -12,14 +12,15 @@ import _root_.doobie.enumerated.Nullability.{Nullable, NoNulls}
 import _root_.doobie.implicits._
 import _root_.doobie.util.fragments
 import java.sql.ResultSet
+import scala.annotation.unchecked.uncheckedVariance
 
 abstract class DoobieMapping[F[_]: Sync](
   val transactor: Transactor[F],
   val monitor:    DoobieMonitor[F]
 ) extends SqlMapping[F] {
 
   type Codec[A]   = (Meta[A], Boolean)
-  type Encoder[A] = (Put[A], Boolean)
+  type Encoder[-A] = (Put[A], Boolean) @uncheckedVariance
```

Comment:
Encoder has to be contravariant or `MappedQuery` has one million pages of type errors.